community/kernel/src/main/java/org/neo4j/kernel/impl/api/store/StorePropertyCursor.java

```diff
@@ -42,17 +42,21 @@
 
     private Lock lock;
 
-    public StorePropertyCursor( PropertyStore propertyStore, Consumer<StorePropertyCursor> instanceCache )
+    public StorePropertyCursor(
+            RecordCursor<PropertyRecord> recordCursor,
+            RecordCursor<DynamicRecord> stringRecordCursor,
+            RecordCursor<DynamicRecord> arrayRecordCursor,
```

Comment:
This guy could alternatively take in the `RecordCursors` object, and then grab the cursors it needs in the constructor.